Scripts/Flag_placer.py

```diff
@@ -0,0 +1,352 @@
+    the read.
+
+    :param cigar: a list containing tuples representing the cigar string.
+    :return overshoot: an integer indicating the number of basepairs in the read before it is mapped.
+    """
+    overshoot = 0
+    for element in cigar:
+        if element[0] != 0:
+            overshoot += element[1]
+        else:
+            break
+
+    return overshoot
+
+
+def issameorientation(read):
+    """ The issameorientation function returns a bool returning true if a pair of reads have the same orientation and
+    the mate is on the same chromosome. It will return false if this is not the case
+
+    :param read: Pysam object containing data of a read.
+    :return bool: A boolean returning True if the reads of a pair have the same orientation.
+    """
+    if read.is_paired and not read.mate_is_unmapped:
+        if read.is_reverse == read.mate_is_reverse and read.reference_name == read.next_reference_name:
+            return True
+
+        else:
+            return False
+
+
+def write_bedfile(flags):
+    """ The write_bedfile function writes a file in BED format that can be loaded in igv and visualises the read data.
+
+    :param flags: a 2d list containing all the flag information.
+    """
+    text = 'track name=Flags description="Flags regions of interest." db=hg19 gffTags=on itemRGB="On"\n'
+
+    for flag in flags:
+        percentage = round(flag[3]['count'] / flag[3]['total'], 2)
+
+        region = f"{flag[0]}\t{flag[1]}\t{flag[2]}"
+        description = f"Name={flag[3]['type']};Readcount={flag[3]['count']};Total_reads={flag[3]['total']};" \
+                      f"Percentage={percentage}"
+
+        if flag[3]['type'] == 'high_insert_size':
+            lengths = flag[3]['lengths']
+            description += f";Avg_insert_size={round(mean(lengths))};Med_insert_size={median(lengths)};" \
+                           f"Lower_limit={min(lengths)};Upper_limit={max(lengths)}"
+
+        if flag[3]['type'] == 'same_orientation':
+            rgb = f"0,192,199"
+
+        elif flag[3]['type'] == 'high_insert_size':
+            rgb = f"232,135,26"
+
+        elif flag[3]['type'] == 'unmapped_mate':
+            rgb = f"218,52,144"
+
+        text += f"{region}\t{description}\t0\t.\t{flag[1]}\t{flag[2]}\t{rgb}\n"
+
+    with open(args.output + f'/{args.name}.bed', 'w') as bedfile:
+        bedfile.write(text)
+
+
+def sort_flags(flags):
+    """ The sort_flags function sorts the flags on starting position using insertionsort.
+
+    :param flags: a 2d list containing all the flag information.
+    :return flags: a 2d list containing all the flag information.
+    """
+    for i in range(1, len(flags)):
+        key = flags[i][1]
+
+        j = i - 1
+        while j >= 0 and key < flags[j][1] and flags[i][0] == flags[j][0]:
+            temp = flags[j+1]
+            flags[j+1] = flags[j]
+            flags[j] = temp
+            j -= 1
+        flags[j + 1][1] = key
+
+    return flags
```

Comment:
Kan je hier niet het volgende gebruiken:

```python
[['test2', 5], ['test3', 10], ['test1', 2]]
sorted(test, key= lambda x: x[1])
[['test1', 2], ['test2', 5], ['test3', 10]]
```

Reply:
Misschien dat het anders kan maar het  probleem is dat die alleen de chromosomen moet sorteren anders plaatst die bijvoorbeeld [chr22, 100] voor [chr1, 200] omdat 100 voor 200 komt maar chromosoom 1 komt eerder dan 22.